core/app/controllers/spree/welcome_page_controller.rb

```diff
@@ -0,0 +1,12 @@
+module Spree
+  class WelcomePageController < Spree::BaseController
+    def index
+      @admin_page_link = Spree::Core::Engine.backend_available? ? "#{current_store.url}:#{request.port}/admin" : nil
```

Comment:
I would make all these URL variables into[ constants](https://www.rubyguides.com/2017/07/ruby-constants/) rather than instance variables. This will make it clear that these values are not going to change, that they will not be different for every instance of the welcome page.

 You can define them at the top of the `WelcomePageController` class, and also add a `.freeze` to the end of the string (e.g. `'https://slack.spreecommerce.org'.freeze`) to make it immutable. Freezing constants helps with performance :)